packages/common/container-definitions/src/deltas.ts

```diff
@@ -61,21 +61,94 @@ export interface IDeltaSender {
     flush(): void;
 }
 
-/** Events emitted by the Delta Manager */
+/**
+ * Events emitted by {@link IDeltaManager}.
+ */
+/* eslint-disable @typescript-eslint/unified-signatures */
 export interface IDeltaManagerEvents extends IEvent {
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `messageBuffer`: TODO
+     */
     (event: "prepareSend", listener: (messageBuffer: any[]) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `message`: TODO
+     */
     (event: "submitOp", listener: (message: IDocumentMessage) => void);
+
+    /**
+     * Emitted immediately after processing an incoming operation (op).
+     *
+     * @remarks Listener parameters:
+     *
+     * - `message`: The op that was processed.
+     *
+     * - `processingTime`: TODO
+     *
+     * @internal
+     */
     (event: "op", listener: (message: ISequencedDocumentMessage, processingTime: number) => void);
+
+    /**
+     * @deprecated Please use the "saved" event instead.
+     */
     (event: "allSentOpsAckd", listener: () => void);
-    (event: "pong" | "processTime", listener: (latency: number) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `latency`: TODO
+     *
+     * @internal
+     */
+    (event: "pong", listener: (latency: number) => void);
+
     /**
-     * The connect event fires once we've received the connect_document_success message from the
-     * server.  This happens prior to the client's join message (if there is a join message).
+     * @deprecated No replacement API recommended.
+     */
+    (event: "processTime", listener: (latency: number) => void);
+
+    /**
+     * Emitted once we've received the connect_document_success message from the server.
+     * This happens prior to the client's join message (if there is a join message).
+     *
+     * @remarks Listener parameters:
+     *
+     * - `details`: TODO
+     *
+     * - `opsBehind`: TODO
      */
     (event: "connect", listener: (details: IConnectionDetails, opsBehind?: number) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
+     *
+     * - `reason`: TODO
+     */
     (event: "disconnect", listener: (reason: string) => void);
+
+    /**
+     * TODO
+     *
+     * @remarks Listener parameters:
```

Comment:
This syntax looks wrong. Should be 
@param <name of argument> - notes

Reply:
TSDoc doesn't currently support drilling down into parameters like that. We could document the "event" and "listener" parameters using `@param` tags here, but those are already documented on `IEvent`. 

An alternative would be to extract `(reason: string) => void` into a type alias and document its parameters on that using `@param`, but that seemed overkill to me.

This structure ends up looking fairly nice in the generated docs, given the limitations. That said, I am 100% open to alternative structure suggestions if you have thoughts.